EssentialFeed/EssentialFeedTests/Feed API/URLSessionHTTPClientTests.swift

```diff
@@ -173,6 +172,11 @@ class URLSessionHTTPClientTests: XCTestCase {
 		}
 		
 		override func startLoading() {
+			if let requestObserver = URLProtocolStub.requestObserver {
+				client?.urlProtocolDidFinishLoading(self)
+				return requestObserver(request)
```

Comment:
I think `return` can be omitted.

Reply:
Hi @mihaicris, we need to return after completing the request so the following code does not execute. Otherwise, we'll end up calling `client?.urlProtocolDidFinishLoading(self)` twice. Calling `urlProtocolDidFinishLoading` twice doesn't seem to break anything, but it's not the correct usage of this API (it may introduce bugs in the future!).

Reply:
Ok, understood the reason for `return`. 

I was a little confused because `requestObserver` closure returns `Void`, which is the same return type as `startLoading()` method. For me it reads a little more clear like this:
```swift
    client?.urlProtocolDidFinishLoading(self)
    requestObserver(request)
    return 
```